contrib/babelfishpg_tsql/src/pl_exec-2.c

```diff
@@ -673,6 +678,10 @@ exec_stmt_exec(PLtsql_execstate *estate, PLtsql_stmt_exec *stmt)
 				need_path_reset = true;
 			}
 	}
+	if (stmt->schema_name != '\0')
+	 	estate->schema_name = stmt->schema_name;
```

Comment:
do we need to rest the db_name and schema_name once exec is done?

Reply:
We don't need to reset the db_name and schema_name for an estate. These will remain unique for an estate.

Reply:
Yes, it's changing for each statement. `estate->db_name` and `estate->schema_name` are reset based on stmt, and inside proc p1 every exec is another stmt and based on which `db_name` and `schema_name` change. We don't need to explicitly change these values.

Reply:
this function will only be called for exec, so exec p1 will set these values but same will carry to insert (insert will not come to this code path), is that expected? exec p2 will reset these to correct values again but there could be no exec statements in-between

Reply:
Yes this is expected. In case of "insert", the parent proc would be p1 and we maintain a different estate for that call. The execution depends on that parent estate->db_name and schema_name which remains unchanged.